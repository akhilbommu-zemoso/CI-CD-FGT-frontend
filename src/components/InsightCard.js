import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import insight from '../assets/insight.png'
import { CardHeader, FTCard } from '../atoms/FTCard';
const useStyles = makeStyles({
    card: {
        width: '300px',
        height: '250px',
        borderRadius: '6px',
        border: '1px,solid,black',
        backgroundColor: 'rgb(255, 255, 255)',
    },
    header: {
        fontSize: '20px',
        fontWeight: 500,
        color: 'black'
    },
    footer: {
        fontSize: '14px',
        textAlign: 'center',
        marginTop: '20px',
    },
    content: {
        textAlign: 'center',
        marginTop: '30px',
        marginBottom: '30px',
        fontSize: '14px',
    },
});
export default function InsightCard(props) {
    const classes = useStyles();
    const header = () => {
        return <CardHeader
            headerTitleStyle={classes.header}
            headerTitleContent={"Insights"}
        />
    }
    return (
        <FTCard
            cardStyle={classes.card}
            contentStyle={classes.content}
            footerStyle={classes.footer}
            header={header()}
            content={<img src={insight} alt="Logo" height='80px' />}
            footer={"No insights are here"}
        />
    );
}
