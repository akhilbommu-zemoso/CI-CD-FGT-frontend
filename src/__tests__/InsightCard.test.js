import React, { Component } from 'react'
import { render } from '@testing-library/react'
import InsightCard from '../components/InsightCard'

it('should take a snapshot', () => {
    const { asFragment } = render(<InsightCard />)
    expect(asFragment(<InsightCard />)).toMatchSnapshot()
});

it('check for header', () => {
    const { getByText } = render(<InsightCard />)
    expect(getByText("Insights")).toBe
});

it('check for content', () => {
    const { getByText } = render(<InsightCard />)
    expect(getByText("No insights are here")).toBe
})
