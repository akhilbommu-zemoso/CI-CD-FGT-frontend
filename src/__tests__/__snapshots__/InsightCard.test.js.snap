// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
      <div>
        <div>
          <p
            class="MuiTypography-root makeStyles-header-2 MuiTypography-body1"
          >
            Insights
          </p>
          <div />
        </div>
        <hr
          class="MuiDivider-root"
          style="margin-top: 10px;"
        />
      </div>
      <p />
      <p
        class="MuiTypography-root makeStyles-content-4 MuiTypography-body1"
      >
        <img
          alt="Logo"
          height="80px"
          src="insight.png"
        />
      </p>
      <p
        class="MuiTypography-root makeStyles-footer-3 MuiTypography-body1"
      >
        No insights are here
      </p>
    </div>
  </div>
</DocumentFragment>
`;
